test/main.mligo

```diff
@@ -0,0 +1,156 @@
+#import "../src/anti.mligo" "ANTI"
+#import "./functions.mligo" "FUNC"
+#import "./contracts/CALLBACK.mligo" "CALLBACK"
+#import "./helpers/errors.mligo" "ERROR"
+#import "./helpers/assert.mligo" "ASSERT"
+
+let test =
+    let () = Test.log("___ TEST SEQUENCE STARTED ___") in
+
+    let random_contract_addr : address = ("KT1MsktCnwfS1nGZmf8QbaTpZ8euVijWdmkC" : address) in
+    let init_token_supply = 777777777777n in
+    let init_token_balance = 1000n in
+
+    let (ant_ctr, ant_addr, cbk_addr, cbk_ctr, alice, bob, james) = FUNC.bootstrap init_token_supply init_token_balance in
+
+    (* Testing GetTotalSupply Callback on Origination *)
+    let test_get_supply_callback_initial = 
+        let () = Test.log("--> test_get_supply_callback_initial : Testing GetTotalSupply Callback on Origination") in
+        let supply_request = ({
+            request =  ();
+            callback = cbk_addr;
+        } : ANTI.getTotalSupply) in
+        let result_cbk = Test.transfer_to_contract ant_ctr (GetTotalSupply supply_request) 0tez in
+        let () = ASSERT.tx_success result_cbk in
+        let cbk_storage = Test.get_storage cbk_ctr in
+        let () = assert(cbk_storage = init_token_supply) in
+        Test.log("OK", cbk_storage)
+    in
+
+    (* Getting Balance of Alice *)
+    let test_get_balance_from_storage =
+        let () = Test.log("--> test_get_balance_from_storage : Alice should have init_token_balance = 1000n") in
+        let retrieved_balance : nat = FUNC.get_balance_from_storage(ant_addr, alice) in
+        let () = assert(retrieved_balance = init_token_balance) in
+        Test.log("OK, Alice :", retrieved_balance)
+    in
+    
+    (* Testing GetBalance Callback for Alice *)
+    let test_get_balance_callback = 
+        let () = Test.log("--> test_get_balance_callback : Testing GetBalance Callback for Alice") in
+        let balance_of_request = ({
+            owner = alice;
+            callback = cbk_addr;
```

Comment:
should be of type `nat contract`

Reply:
The callback actually renders an error on my side if I use the `nat contract` type
I don't find any use of this type on the docs
Do you have an example on hand ?

Reply:
I don't know maybe there is a problem in the contract, but it the type appearing here https://github.com/smart-chain-fr/mechanical-token/blob/main/src/anti.mligo#L25